kubernetes/cmsweb/scripts/deploy-secrets.sh

```diff
@@ -42,13 +42,27 @@ conf=$3
         	cp $cmsweb_crt $conf/frontend-ds/hostcert.pem
     	fi
     fi
-
-
 	secretdir=$conf/$srv
         # the underscrore is not allowed in secret names
         osrv=$srv
         srv=`echo $srv | sed -e "s,_,,g"`
         files=""
+
+    if [ "$srv" == "auth-proxy-server" ] || [ "$srv" == "x509-proxy-server" ] || [ "$srv" == "scitokens-proxy-server" ] ; then
+        if [ -d $secretdir ] && [ -n "`ls $secretdir`" ] && [ -f $secretdir/client.secrets ]; then
+           export CLIENT_SECRET=`grep CLIENT_SECRET $secretdir/client.secrets | head -n1 | awk '{print $2}'`
+           export CLIENT_ID=`grep CLIENT_ID $secretdir/client.secrets | head -n1 | awk '{print $2}'`
+           export IAM_CLIENT_ID=`grep IAM_CLIENT_ID $secretdir/client.secrets | head -n1 | awk '{print $2}'`
+           export IAM_CLIENT_SECRET=`grep IAM_CLIENT_SECRET $secretdir/client.secrets | head -n1 | awk '{print $2}'`
+	   if [ -f $secretdir/config.json ]; then
+              echo "$(jq '.client_id=env.CLIENT_ID' $secretdir/config.json)" > $secretdir/config.json
```

Comment:
I would avoid using `jq` tool since it may not present on a node. Instead it is much safer to rely on `sed`. For instance, you can replace a secret as following:
```
sed -i -e "s,CLIENT_ID,$CLIENT_ID,g" $secretdir/config.json
```
etc.

Here we only need to ensure that our `config.json` has these lines:
```
    "client_id": "CLIENT_ID",
    "client_secret": "CLIENT_SECRET",
    "iam_client_id": "IAM_CLIENT_ID",
    "iam_client_secret": "IAM_CLIENT_SECRET",
...
    "http://COUCHDB_USER:COUCHDB_PASSWORD@,,,,"
```

So, in configs we will use values for our secret attributes as name of env variables which will hold the actual values.